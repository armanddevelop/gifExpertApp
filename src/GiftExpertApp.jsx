import { useState } from "react";
import { AddCategory, GiftGrid } from "./components/";

export const GiftExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);
  return (
    <>
      <h1>GiftExpertApp</h1>
      <AddCategory setCategories={setCategories} />
      <hr />
      {categories.map((category) => (
        <GiftGrid category={category} key={category} />
      ))}
    </>
  );
};
