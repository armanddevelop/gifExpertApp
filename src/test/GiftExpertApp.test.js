import { GiftExpertApp } from "../GiftExpertApp";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import toJson from "enzyme-to-json";

describe("Test to GiftExpertApp component", () => {
  it("snapshot for GiftExpertApp", () => {
    const wrapper = shallow(<GiftExpertApp />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should show GiftGrid component", () => {
    const categories = ["one punch"];
    const wrapper = shallow(<GiftExpertApp defaultCategories={categories} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find("GiftGrid").length).toBe(categories.length);
  });
});
