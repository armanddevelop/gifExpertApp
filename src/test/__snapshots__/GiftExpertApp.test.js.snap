// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to GiftExpertApp component should show GiftGrid component 1`] = `
<Fragment>
  <h1>
    GiftExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GiftGrid
    category="one punch"
    key="one punch"
  />
</Fragment>
`;

exports[`Test to GiftExpertApp component snapshot for GiftExpertApp 1`] = `
<Fragment>
  <h1>
    GiftExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
